src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -0,0 +1,2293 @@
+
+  Inputs:
+    type - type of the function called
+    loc - location in the program
+    symbol_table - symbol table
+
+ Outputs: code for the StringBuilder.append(Object) function:
+          > string1 = arguments[1].toString()
+          > string_expr1 = string_to_string_expr(string1)
+          > string_expr2 = concat(this, string_expr1)
+          > this = string_expr_to_string(string_expr2)
+          > return this
+
+\*******************************************************************/
+
+codet java_string_library_preprocesst::make_string_builder_append_object_code(
+  const code_typet &type,
+  const source_locationt &loc,
+  symbol_tablet &symbol_table)
+{
+  code_typet::parameterst params=type.parameters();
+  exprt this_obj=symbol_exprt(params[0].get_identifier(), params[0].type());
+
+  // Code to be returned
+  code_blockt code;
+
+  // String expression that will hold the result
+  string_exprt string_expr1=fresh_string_expr(loc, symbol_table, code);
+
+  exprt::operandst arguments=process_parameters(
+    type.parameters(), loc, symbol_table, code);
+  assert(arguments.size()==2);
+
+  // > string1 = arguments[1].toString()
+  exprt string1=fresh_string(this_obj.type(), loc, symbol_table);
+  code_function_callt fun_call;
+  fun_call.lhs()=string1;
+  // TODO: we should look in the symbol table for such a symbol
+  fun_call.function()=symbol_exprt(
+    "java::java.lang.Object.toString:()Ljava/lang/String;");
+  fun_call.arguments().push_back(arguments[1]);
+  code_typet fun_type;
+  fun_type.return_type()=string1.type();
+  fun_call.function().type()=fun_type;
+  code.copy_to_operands(fun_call);
+
+  // > string_expr1 = string_to_string_expr(string1)
+  code.copy_to_operands(code_assign_java_string_to_string_expr(
+    string_expr1, string1, symbol_table));
+
+  // > string_expr2 = concat(this, string1)
+  exprt::operandst concat_arguments(arguments);
+  concat_arguments[1]=string_expr1;
+  string_exprt string_expr2=string_expr_of_function_application(
+    ID_cprover_string_concat_func, concat_arguments, loc, symbol_table, code);
+
+  // > this = string_expr
+  code.copy_to_operands(code_assign_string_expr_to_java_string(
+    this_obj, string_expr2, symbol_table));
+
+  // > return this
+  code.copy_to_operands(code_returnt(this_obj));
+
+  return code;
+}
+
+/*******************************************************************\
+
+Function: java_string_library_preprocesst::make_equals_code
+
+  Inputs:
+    type - type of the function call
+    loc - location in the program_invocation_name
+    symbol_table - symbol table
+
+  Outputs: Code corresponding to:
+    > string_expr1 = {this->length; *this->data}
+    > string_expr2 = {arg->length; *arg->data}
+    > return cprover_string_equal(string_expr1, string_expr2)
+
+  Purpose: Used to provide code for the Java String.equals(Object) function.
+
+\*******************************************************************/
+
+codet java_string_library_preprocesst::make_equals_code(
```

Comment:
Similarly consider naming this equals_function instead of equals to disambiguate the method from the operator(s)